### ISSUE TITLE
Basic auth support suggestion

### ISSUE THREAD

**Issue by shvez** [microsoft/perfview#1885]
Hi, guys

just recently we started to get benefits from usage of your great tool. Thank you for it
There is one thing that does not work for us well. We keep our symbols on our own server that is protected by basic auth. So, our symbol store url looks like this: https://user:pass@reset.of.url.  PerfView can not download symbols from our servers. that is sad.

I started to look into code and found out that you have some support for basic auth in your AuthToken. So, I added very simple handler that works with basic auth. Code is below.

Is it possible to add support for BasicAuth so that I do not have to merge this change once we use new version? I hope it does not add too much complexity and consistent enough to be included

```
    internal sealed class BasicAuthHandler : SymbolReaderAuthHandlerBase
    {
        /// <summary>
        /// Prefix to put in front of logging messages.
        /// </summary>
        private const string LogPrefix = "BasicAuthHanler: ";

        public BasicAuthHandler(TextWriter log) : base(log, LogPrefix)
        {
        }

        private static readonly char[] delimiter = { ':' };
        protected override bool TryGetAuthority(Uri requestUri, out Uri authority)
        {
            if (string.IsNullOrEmpty(requestUri.UserInfo) || !requestUri.UserInfo.Contains(":"))
            {
                authority = null;
                return false;
            }
            authority = requestUri;
            return true;
        }

        protected override Task<AuthToken?> GetAuthTokenAsync(RequestContext context, SymbolReaderHandlerDelegate next, Uri authority,
            CancellationToken cancellationToken)
        {

            var strings = authority.UserInfo.Split(delimiter, StringSplitOptions.RemoveEmptyEntries);
            if (strings.Length < 2)
            {
                return base.GetAuthTokenAsync(context, next, authority, cancellationToken);
            }

            WriteStatusLog("auth token for basic auth provided");

            var token = AuthToken.CreateBasicFromUsernameAndPassword(strings[0], strings[1]);
            return Task.FromResult<AuthToken?>(token);
        }
    }
```

**Comment by pharring** [comment]
Hi @shvez,
I added auth token support for the other providers and I think this would be a great addition. Please feel free to create a Pull Request for it and I'll take a look. I don't have a good way to test it myself.

**Comment by shvez** [comment]
@pharring I added menu option and updated documentation. I have to admit that I'm not a native speaker. text might require some rewording

**Comment by pharring** [comment]
Fixed in #1887 
@brianrob please close (I don't have permission in this repo)
